test/core/client_config/resolvers/sockaddr_resolver_test.c

```diff
@@ -50,9 +67,19 @@ static void test_succeeds(grpc_resolver_factory *factory, const char *string) {
   args.uri = uri;
   resolver = grpc_resolver_factory_create_resolver(factory, &args);
   GPR_ASSERT(resolver != NULL);
+
+  on_resolution_arg on_res_arg;
+  memset(&on_res_arg, 0, sizeof(on_res_arg));
+  on_res_arg.expected_server_name = gpr_strdup(uri->path);
```

Comment:
I think you could avoid copying this string (and freeing it below).  Since `uri` is still in-scope until after this test finishes, it should be fine to use it directly.